cypress-tests/cypress/commands/platform/platformApiCommands.js

```diff
@@ -191,34 +182,89 @@ Cypress.Commands.add("apiGetGroupId", (groupName) => {
+                        const ds = response.body.data_sources.find(
+                            (d) => d.name.toLowerCase().trim() === normalizedSearchName
+                        );
+                        return ds?.id;
+                    })
+                    .filter(Boolean);
+
+                return dsIds;
+            });
+    });
+});
+
+Cypress.Commands.add("apiGetAppIdByName", (appName) => {
+    return cy.getAuthHeaders().then((headers) => {
+        return cy
+            .request({
+                method: "GET",
+                url: `${Cypress.env("server_host")}/api/apps`,
+                headers: headers,
+                log: false,
+            })
+            .then((response) => {
+                expect(response.status).to.equal(200);
+                const app = response.body.apps.find((app) => app.name === appName);
+                expect(app, `App with name "${appName}" not found`).to.exist;
+                return app.id;
+            });
+    });
+});
+
 Cypress.Commands.add("apiUpdateUserRole", (email, role) => {
-    return cy.task("dbConnection", {
-        dbconfig: Cypress.env("app_db"),
-        sql: `
+    return cy
+        .task("dbConnection", {
+            dbconfig: Cypress.env("app_db"),
+            sql: `
       SELECT id 
       FROM users
       WHERE email='${email}'
       LIMIT 1;
     `,
-    }).then((resp) => {
-        const userId = resp.rows[0]?.id;
-        if (!userId) throw new Error(`User with email ${email} not found`);
-        return userId;
-    }).then((userId) => {
-        return cy.getAuthHeaders().then((headers) => {
-            return cy.request({
-                method: "PUT",
-                url: `${Cypress.env("server_host")}/api/v2/group-permissions/role/user`,
-                headers: headers,
-                body: {
-                    newRole: role,
-                    userId: userId,
-                },
-            }).then((response) => {
-                expect(response.status).to.equal(200);
+        })
+        .then((resp) => {
+            const userId = resp.rows[0]?.id;
+            if (!userId) throw new Error(`User with email ${email} not found`);
+            return userId;
+        })
+        .then((userId) => {
+            return cy.getAuthHeaders().then((headers) => {
+                return cy
+                    .request({
+                        method: "PUT",
+                        url: `${Cypress.env("server_host")}/api/v2/group-permissions/role/user`,
+                        headers: headers,
+                        body: {
+                            newRole: role,
+                            userId: userId,
+                        },
+                    })
+                    .then((response) => {
+                        expect(response.status).to.equal(200);
+                    });
             });
         });
-    });
 });
 
 Cypress.Commands.add(
```

Comment:
I think this command we need to make more simple, it is hard to understand,
I think wer can make this more readable, so that it will be easy for anyone to make future updated if needed.